src/otx/algo/detection/ops/nms.py

```diff
@@ -219,3 +221,220 @@ def batched_nms(
+        (batch_inds == batch_template.unsqueeze(1)).unsqueeze(-1),
+        batched_dets.new_zeros(1),
+    )
+
+    batched_labels = cls_inds.unsqueeze(0).repeat(batch_size, 1)
+    batched_labels = batched_labels.where((batch_inds == batch_template.unsqueeze(1)), batched_labels.new_ones(1) * -1)
+
+    batch_size = batched_dets.shape[0]
+
+    # expand tensor to eliminate [0, ...] tensor
+    batched_dets = torch.cat((batched_dets, batched_dets.new_zeros((batch_size, 1, 5))), 1)
+    batched_labels = torch.cat((batched_labels, batched_labels.new_zeros((batch_size, 1))), 1)
+    if output_index and pre_inds is not None:
+        # batch all
+        pre_inds = pre_inds[batch_inds, box_inds]
+        pre_inds = pre_inds.unsqueeze(0).repeat(batch_size, 1)
+        pre_inds = pre_inds.where((batch_inds == batch_template.unsqueeze(1)), pre_inds.new_zeros(1))
+        pre_inds = torch.cat((pre_inds, -pre_inds.new_ones((batch_size, 1))), 1)
+    # sort
+    is_use_topk = keep_top_k > 0 and (torch.onnx.is_in_onnx_export() or keep_top_k < batched_dets.shape[1])
+    if is_use_topk:
+        _, topk_inds = batched_dets[:, :, -1].topk(keep_top_k, dim=1)
+    else:
+        _, topk_inds = batched_dets[:, :, -1].sort(dim=1, descending=True)
+    topk_batch_inds = torch.arange(batch_size, dtype=topk_inds.dtype, device=topk_inds.device).view(-1, 1)
+    batched_dets = batched_dets[topk_batch_inds, topk_inds, ...]
+    batched_labels = batched_labels[topk_batch_inds, topk_inds, ...]
+    if output_index:
+        if pre_inds is not None:
+            topk_inds = pre_inds[topk_batch_inds, topk_inds, ...]
+        return batched_dets, batched_labels, topk_inds
+    # slice and recover the tensor
+    return batched_dets, batched_labels
+
+
+class ONNXNMSop(torch.autograd.Function):
+    """Create onnx::NonMaxSuppression op.
+
+    NMS in mmcv only supports one class with no batch info. This class assists
+    in exporting NMS of ONNX's definition.
+    """
+
+    @staticmethod
+    def forward(
+        ctx,  # noqa: ANN001, ARG004
+        boxes: Tensor,
+        scores: Tensor,
+        max_output_boxes_per_class: int,
+        iou_threshold: float,
+        score_threshold: float,
+    ) -> Tensor:
+        """Get NMS output indices.
+
+        Args:
+            ctx (Context): The context with meta information.
+            boxes (Tensor): The bounding boxes of shape [N, num_boxes, 4].
+            scores (Tensor): The detection scores of shape
+                [N, num_boxes, num_classes].
+            max_output_boxes_per_class (int): Maximum number of output
+                boxes per class of nms.
+            iou_threshold (float): IOU threshold of nms.
+            score_threshold (float): score threshold of nms.
+
+        Returns:
+            Tensor: Selected indices of boxes. 2-D tensor of shape
+            (num_selected_indices, 3) with each row of
+            [batch_index, class_index, box_index].
+        """
+        from mmcv.ops import nms
```

Comment:
Isn't getting nms in this PR's scope?

Reply:
Yes it is. Changed by [3f62054](https://github.com/openvinotoolkit/training_extensions/pull/3366/commits/3f62054242114aea15660194ea6790d5f68aa851)